dispatcher/dispatcher.go

```diff
@@ -0,0 +1,128 @@
+package dispatcher
+
+import (
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	dispatch "github.com/textileio/go-textile-core/dispatcher"
+)
+
+// Token is a simple unique ID used to reference a registered callback.
+type Token uint
+
+// lastID is the last ID used by the singleton Dispatcher.
+var lastID uint
+
+const (
+	// ErrPersistence means that a write to the underlying event store failed.
+	ErrPersistence = esError("persistance failure")
+	// ErrTokenNotFound means the given token was not found in the dispatcher's reducer map.
+	ErrTokenNotFound = esError("token not found")
+)
+
+type esError string
+
+func (e esError) Error() string { return string(e) }
```

Comment:
Are you using this style of error defining compared to `errors.New()` for some reason?

Reply:
I read about it here: https://dave.cheney.net/2016/04/07/constant-errors

Reply:
I supposed ;), that's a kind of well-known discussion in the Go world. To be honest, I'm from the school that thinks that if you include a malicious party dependency in your code, you'll be screwed up anyway. (as per `io.EOF` isn't constant example)

Reply:
So then are you advocating to just use something like the following?
```
// ErrPersistence means that a write to the underlying event store failed.
ErrPersistence = errors.New("persistence failure")
```

Reply:
Also noticed a spelling error there

Reply:
Only saying that's a usual idiomatic way to define sentinel errors. Your implementation based on Dave's argument is totally correct from the standpoint of const vs var... so, no strong claim here.